xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -198,6 +198,35 @@ impl EncryptedMessageStore {
         crypto_utils::rng().fill_bytes(&mut key[..]);
         key
     }
+
+    pub fn release_connection(&self) -> Result<(), StorageError> {
+        let conn = self.raw_conn()?;
+
+        // Explicitly drop the connection to release the lock
+        drop(conn);
```

Comment:
Looking through the [ConnectionManager](https://docs.rs/diesel/latest/diesel/r2d2/struct.ConnectionManager.html) Documentation, we can safely hold onto this and just drop + re-create the pool whenever we need to. The connection manager does not hold any connections to the database itself, but it allows creating them.

So, we can refactor the `EncryptedMessageStore` api a little bit to make this possible. 

- wrap `Pool` in an Option
- add a field to `EncryptedMessageStore` for `ConnectionManager`, so that we don't have to read from disk for the database file
- in `release_connection`, use [Option::take](https://doc.rust-lang.org/std/option/enum.Option.html#method.take) to replace the `Pool` with `None`
- `reconnect` stays largely the same, except you can use the already-stored `ConnectionManager` on `EncryptedStore`


You probably need to bubble this API up to `Client`/`client.rs` and then you can get rid of the global `STORE_MANAGER`

this should probably work

happy to pair with you on this if you want!

Reply:
I'd love the pair.